Tests/js/views/services/assets/ez-subitemboxviewservice-tests.js

```diff
@@ -0,0 +1,166 @@
+            this.contentType = {};
+            this.location = {};
+            this.locationId = 42;
+            this.searchResultList = [{content: this.content, contentType: this.contentType, location: this.location}];
+            this.request = {
+                params: {
+                    id: this.locationId,
+                    searchString: this.searchString,
+                    limit: this.limit
+                }
+            };
+            this.app = new Y.Base();
+
+            this.service = new Y.eZ.SubitemBoxViewService({
+                request: this.request,
+                app: this.app,
+                response: {},
+            });
+
+            this.service.search = new Mock();
+        },
+
+        tearDown: function () {
+            this.service.destroy();
+            delete this.service;
+        },
+        
+        "Should set the content, contentType and location": function () {
+            var callbackCalled = false;
+
+            Mock.expect(this.service.search, {
+                method: 'findContent',
+                args: [Mock.Value.Object, Mock.Value.Function],
+                run: Y.bind(function (search, callback) {
+                    Assert.isTrue(search.loadLocation, 'Should load location');
+                    Assert.isTrue(search.loadContentType, 'Should load contentType');
+                    Assert.areSame(search.limit, 1, 'Search limit should be 1');
+                    Assert.areSame(search.offset, 0, 'Search offset should be 0');
+                    Assert.areSame(search.viewName, 'location-' + this.locationId, 'View name should contain the location id');
+                    Assert.areSame(search.query.LocationIdCriterion, this.locationId, 'Query should have a correct location id criterion');
+
+                    callback(false, this.searchResultList);
+                }, this),
+            });
+            this.service.load(function(){
+                callbackCalled = true;
+            });
+
+            Assert.areSame(
+                this.content,
+                this.service.get('content'),
+                "The content should be setted"
+            );
+
+            Assert.areSame(
+                this.location,
+                this.service.get('location'),
+                "The location should be setted"
+            );
+
+            Assert.areSame(
+                this.contentType,
+                this.service.get('contentType'),
+                "The contentType should be setted"
+            );
+
+            Assert.isTrue(callbackCalled, 'Callback should have been called');
+        },
+
+        "Should handle search error ": function () {
+            var callbackCalled = false,
+                errorCalled = false;
+
+            this.service.on('error', function (e) {
+                Y.Assert.isObject(e, "An event facade should be provided");
+                Assert.isString(e.message, "The message property should be filled");
```

Comment:
actually you should check the value of that which should be something like `subitem.error.loading.list domain=subitem`